src/Profiles.ts

```diff
@@ -143,130 +134,291 @@ export class Profiles {
+                    vscode.window.showInformationMessage(localize("createNewConnection.rejectUnauthorize","Operation Cancelled"));
+                    return undefined;
+                }
+                schemaValues[value] = rejectUnauthorize;
+                break;
+            default:
+                const description: string = schema[value].optionDefinition.description.toString();
+                switch (schema[value].type) {
+                    case "string":
+                        options = await this.optionsValue(value, schema);
+                        const profValue = await vscode.window.showInputBox(options);
+                        if (profValue === "") {
+                            break;
+                        }
+                        schemaValues[value] = profValue;
+                        break;
+                    case "boolean":
+                        const quickPickBooleanOptions: vscode.QuickPickOptions = {
+                            placeHolder: description,
+                            ignoreFocusOut: true,
+                            canPickMany: false
+                        };
+                        const selectBoolean = ["True", "False"];
+                        const chosenValue = await vscode.window.showQuickPick(selectBoolean, quickPickBooleanOptions);
+                        if (chosenValue === selectBoolean[0]) {
+                                isTrue = true;
+                            } else if (chosenValue === selectBoolean[1]) {
+                                isTrue = false;
+                            } else {
+                                vscode.window.showInformationMessage(localize("createNewConnection","Operation Cancelled"));
+                                return undefined;
+                            }
+                        schemaValues[value] = isTrue;
+                        break;
+                    case "number":
+                        options = await this.optionsValue(value, schema);
+                        const enteredValue = await vscode.window.showInputBox(options);
+                        if (!Number.isNaN(Number(enteredValue))) {
+                            schemaValues[value] = Number(enteredValue);
+                            } else {
+                                if (schema[value].optionDefinition.hasOwnProperty("defaultValue")){
+                                    schemaValues[value] = schema[value].optionDefinition.defaultValue;
+                                } else {
+                                    schemaValues[value] = undefined;
+                                }
+                            }
+                        break;
+                    default:
+                        options = await this.optionsValue(value, schema);
+                        const defaultValue = await vscode.window.showInputBox(options);
+                        switch (defaultValue){
+                            case "True" || "true":
```

Comment:
Why are Boolean values parsed here, since in the `default` case the type of the profile field is not "boolean"? Sorry, I'm probably missing something 😢

Reply:
there is a type of `type: -boolean -null`,`type: -string -null`, and/or `type: -number -null` in the zftp cli plugin 
![image](https://user-images.githubusercontent.com/49491949/80425212-3780d380-88b1-11ea-9172-1562b7790075.png)